docs/howto/architectures.rst

```diff
@@ -2,19 +2,57 @@ Architectures
 =============
 
 By default, Snapcraft builds a snap to run on the same architecture as the build
-environment. This behaviour can be modified with the root keyword
-``architectures`` in the snap’s ``snapcraft.yaml``.
+environment. This behaviour can be modified with the top-level keywords
+``architectures`` and ``platforms`` in the snap’s ``snapcraft.yaml``.
 
-The ``architectures`` keyword is used to create a build plan. See
-:ref:`build plans<build-plans>` for an explanation on how build plans are
-created.
+The ``architectures`` and ``platforms`` keywords are used to create a build
+plan. See :ref:`build plans<build-plans>` for an explanation on how build
+plans are created.
+
+The keywords are base-dependent:
+
+* ``platforms`` is used for ``core24`` snaps
+* ``architectures`` is used for ``core20`` and ``core22`` snaps
 
 How to create a snap for a specific architecture
 ------------------------------------------------
 
 To create a snap that will be built on ``amd64`` and built for ``amd64``, use
 one of the ``snapcraft.yaml`` snippets below.
 
+core24
+^^^^^^
+
+.. code-block:: yaml
+
+  platforms:
+    amd64:
+      build-on: [amd64]
+      build-for: [amd64]
+
+Building on ``amd64`` will produce one snap built for ``amd64``. If
+``build-for`` is omitted, it will assume the platform name. The following
```

Comment:
What happens if you leave out `build-for` but call the platform something invalid?

Reply:
Ack, good catch.  I clarified that and caught a mistake!